### PR TITLE
Rework audit_deps to use full advisory IDs

### PR DESCRIPTION
**Pull request by diracdeltas**
Fix https://github.com/brave/brave-browser/issues/19592
Also remove support for auditing prod deps only since it turns out we don't really separate prod deps in package.json.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. audit_deps should pass
2. audit_deps should fail if any entries in https://github.com/brave/brave-core/pull/11248/files#diff-a69e633bdbae5d870abc8b9c5b0086be91e75569ea3526810d107598d6df478fR35 are removed

i also tested this with npm@8 and it seems to work

### CODE REVIEW

**Review by wknapik — APPROVED** [review]

**Review by wknapik — APPROVED** [review]

**Review by iefremov — APPROVED** [review]
unlocking DEPS

File: script/audit_deps.py [hunk 1]
```diff
@@ -19,8 +19,7 @@
 # Use all (sub)paths except these for npm audit.
 NPM_EXCLUDE_PATHS = [
     'build',
-    os.path.join('node_modules'),
-    os.path.join('vendor', 'brave-extension', 'node_modules'),
+    os.path.join('node_modules')
```
Comment (wknapik) [inline script/audit_deps.py:22]: Could just be `'node_modules'`.

File: script/audit_deps.py [hunk 2]
```diff
@@ -34,39 +33,41 @@
...
+    'https://github.com/advisories/GHSA-w8qv-6jwh-64r5',  # rxdos
+    'https://github.com/advisories/GHSA-ww39-953v-wcq6',  # rxdos
+    'https://github.com/advisories/GHSA-whgm-jr23-g3j9'   # rxdos
 ]
 
 
 def main():
     """Audit a specified path, or the whole project."""
 
+    if len(IGNORED_NPM_ADVISORIES) > 0:
```
Comment (wknapik) [inline script/audit_deps.py:49]: Could just be `if IGNORED_NPM_ADVISORIES:`.

File: script/audit_deps.py [hunk 3]
```diff
@@ -34,39 +33,41 @@
...
+        print(
+            f"Ignoring NPM advisories "
+            f"{', '.join(map(str, IGNORED_NPM_ADVISORIES))}"
+        )
+    if len(IGNORED_CARGO_ADVISORIES) > 0:
```
Comment (wknapik) [inline script/audit_deps.py:54]: Could just be `if IGNORED_CARGO_ADVISORIES:`.

**Review by wknapik — APPROVED** [review]
Added some comments, but they're not super important, let's get this merged and uplifted.

**Pull request merged**